From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Helvetica Volubi <suisuroru@blue-millennium.fun>
Date: Mon, 3 Feb 2025 16:51:01 +0800
Subject: [PATCH] Leaves: Syncmatica Protocol

This patch is Powered by Syncmatica(https://github.com/End-Tech/syncmatica)
Co-authored by: violetc <58360096+s-yh-china@users.noreply.github.com>
As a part of : Leaves (https://github.com/LeavesMC/Leaves)
Licensed under: GPL-3.0 (https://www.gnu.org/licenses/gpl-3.0.html)

diff --git a/net/minecraft/server/network/ServerGamePacketListenerImpl.java b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
index fb64239bc0872a4254ebe5f94bcc16b72e56207c..c2b63df16c037dc23d0970f0869659df2153064b 100644
--- a/net/minecraft/server/network/ServerGamePacketListenerImpl.java
+++ b/net/minecraft/server/network/ServerGamePacketListenerImpl.java
@@ -356,9 +356,12 @@ public class ServerGamePacketListenerImpl
         this.chatMessageChain = new FutureChain(server.chatExecutor); // CraftBukkit - async chat
         this.commandSuggestionsProvider = new ServerCommandSuggestionsProvider(player);
         this.tickEndEvent = new io.papermc.paper.event.packet.ClientTickEndEvent(player.getBukkitEntity()); // Paper - add client tick end event
+        this.exchangeTarget = new org.leavesmc.leaves.protocol.syncmatica.exchange.ExchangeTarget(this); // Leaves - Syncmatica Protocol
         this.playerGameConnection = new io.papermc.paper.connection.PaperPlayerGameConnection(this); // Paper
     }
 
+    public final org.leavesmc.leaves.protocol.syncmatica.exchange.ExchangeTarget exchangeTarget; // Leaves - Syncmatica Protocol
+
     // Paper start - configuration phase API
     @Override
     public io.papermc.paper.connection.PlayerCommonConnection getApiConnection() {
